File: Backend/TravelPlanner.Core/DomainModels/Flight.cs
using System;
using System.Runtime.Serialization;

namespace TravelPlanner.Core.DomainModels
{
    [DataContract]
    public class Flight
    {
        [DataMember]
        public string FlightId { get; set; } 

        [DataMember]
        public FlightDuration FlightDuration { get; set; }

        [DataMember]
        public AirportFlightStatus Departure { get; set; }

        [DataMember]
        public AirportFlightStatus Arrival { get; set; }

        [DataMember]
        public string AirlineId { get; set; }
        
        [DataMember]
        public string FlightNumber { get; set; }
    }
}

File: TravelPlannerApp/app/src/main/java/com/travelplanner/models/Flight.kt
package com.travelplanner.models

import java.time.LocalDateTime

class Flight(val flightId: String, val flightDuration: FlightDuration,
             val departure: AirportFlightStatus, val arrival: AirportFlightStatus,
             val airlineId: String, val flightNumber: String)

class FlightDuration(val days: Int, val hours: Int, val minutes: Int)

class AirportFlightStatus(val airportCode: String, val scheduledTimeLocal: LocalDateTime, val terminalName: String)